### DISCUSSION TITLE
unable to clear sort column in data provider

### DISCUSSION THREAD

**Discussion by dncpax** [Budibase/budibase#2947] [Help] [answered]
Hi there.
It may be just me, but I can't find a way to clear the sorting of a datasource (in this case it's a postgresql)... once set, can't unset?
![image](https://user-images.githubusercontent.com/2836636/136675639-a15eb75c-1576-493f-a3a8-cacac014fc89.png)

I can't clear the sort column dropdown...

Any tips?

Thanks.

**Comment by dncpax** [comment]
on a related note, I can't also sort by number... it always sorts alphabetically... even if it's a number column... that's why I'm trying to unset the sorting so I can do it on the query.

**Comment by mjashanks** [comment, marked as answer]
Thanks for reporting this - I've logged the alphabetical sorting issue here: https://github.com/Budibase/budibase/issues/2962

It's best to sort inside the query anyway. 

I can suggest a hacky workaround. 

1. Select back a dummy alias column to "Sort" by... e.g. 
`select *, '0' as MyCustomSort from public."customers" order by numberColumn`
2. In your dataprovider, set the `Sort Column` to `MyCustomSort`

This way, the `Sort Column` does not change the order from the returned data set

**Reply by dncpax** [reply]
hack applied and working!

**Reply by hawkinsjon** [reply]
Thank you for this @mjashanks. I was running into a similar issue solved by this workaround.
